shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java

```diff
@@ -68,20 +66,24 @@ public int run(CommandLine cl) throws IOException {
     NetAddress address = QuorumCommand.stringToAddress(serverAddress);
 
     jmClient.transferLeadership(address);
-
-    MasterInquireClient inquireClient = MasterInquireClient.Factory
-            .create(mConf, FileSystemContext.create(mConf).getClientContext().getUserState());
     boolean success = true;
     // wait for confirmation of leadership transfer
     try {
       CommonUtils.waitFor("Waiting for leadership transfer to finalize", () -> {
-        InetSocketAddress leaderAddress;
         try {
-          leaderAddress = inquireClient.getPrimaryRpcAddress();
+          GetQuorumInfoPResponse quorumInfo = jmClient.getQuorumInfo();
+
+          Optional<QuorumServerInfo>
+                  leadingMasterInfoOpt = quorumInfo.getServerInfoList().stream()
+                  .filter(QuorumServerInfo::getIsLeader).findFirst();
+          String leadingMasterAddr = leadingMasterInfoOpt.isPresent()
+                  ? netAddressToString(leadingMasterInfoOpt.get().getServerAddress()) : "UNKNOWN";
+          return leadingMasterAddr.equals(netAddressToString(address));
```

Comment:
why convert to `String`?
`NetAddress`es should be comparable.

Reply:
Sure, @codings-dan you can use `NetAddress#equals` to compare these two address are the same.